docs/spark.md

```diff
@@ -0,0 +1,126 @@
+...   return hvd.rank()
+...
+>>> import horovod.spark
+>>> horovod.spark.run(fn, args=(42,))
+Running 16 processes...
+[Stage 0:>                                                        (0 + 16) / 16]
+Hello, rank = 15, local_rank = 3, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 13, local_rank = 1, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 8, local_rank = 0, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 9, local_rank = 1, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 10, local_rank = 2, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 11, local_rank = 3, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 6, local_rank = 2, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 4, local_rank = 0, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 0, local_rank = 0, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 1, local_rank = 1, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 2, local_rank = 2, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 5, local_rank = 1, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 3, local_rank = 3, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 12, local_rank = 0, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 7, local_rank = 3, size = 16, local_size = 4, magic_number = 42
+Hello, rank = 14, local_rank = 2, size = 16, local_size = 4, magic_number = 42
+[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
+>>>
+```
+
+### End-to-end example
+
+[keras_spark_rossmann.py script](../examples/keras_spark_rossmann.py) provides
+an example of end-to-end data preparation and training of a model for the
+[Rossmann Store Sales](https://www.kaggle.com/c/rossmann-store-sales) Kaggle
+competition.
+
+It is inspired by an article [An Introduction to Deep Learning for Tabular Data](https://www.fast.ai/2018/04/29/categorical-embeddings/)
+and leverages the code of the notebook referenced in the article.
+
+The example is split into three parts:
+1. The first part performs complicated data preprocessing over initial set
+of CSV files provided by competition and gathered by the community.
+2. The second part defines a Keras model and performs a distributed training
+of the model using Horovod in Spark.
+3. The third part performs prediction using the best model and creates
+a submission file.
+
+In order to run the example, please install the following dependencies:
+* `pyspark`
+* `petastorm >= 0.5.1`
+* `h5py >= 2.9.0`
+* `tensorflow-gpu >= 1.12.0` (or `tensorflow >= 1.12.0`)
+* `horovod >= 0.15.3`
+
+Run the example:
+```bash
+$ wget https://raw.githubusercontent.com/uber/horovod/master/examples/keras_spark_rossmann.py
+$ wget http://files.fast.ai/part2/lesson14/rossmann.tgz
+$ tar zxvf rossmann.tgz
+$ python keras_spark_rossmann.py
+```
+
+### Spark cluster setup
```

Comment:
@felixcheung, do you have better recommendations for users to set up GPU Spark cluster at this point?